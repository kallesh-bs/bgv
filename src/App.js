import React from 'react';
import 'styles/App.css';
import { MainRoutes } from './routes';
import "language/i18n";

function App() {
  return <MainRoutes/>;
}

export default App;
