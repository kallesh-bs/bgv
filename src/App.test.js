/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

// Mock the MainRoutes component
jest.mock('./routes', () => ({
  MainRoutes: () => <div data-testid="main-routes">Mocked MainRoutes</div>,
}));

// eslint-disable-next-line no-undef
describe('App Component', () => {
  it('renders without crashing', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders MainRoutes component', () => {
    const tree = renderer.create(<App />);
    expect(tree.root.findByProps({ 'data-testid': 'main-routes' }).children).toEqual(['Mocked MainRoutes']);
  });
});
