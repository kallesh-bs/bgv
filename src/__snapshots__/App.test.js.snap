// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders without crashing 1`] = `
<div
  data-testid="main-routes"
>
  Mocked MainRoutes
</div>
`;
